core/templates/dev/head/pages/exploration_player/PlayerServices.js

```diff
@@ -369,9 +374,19 @@ oppia.factory('oppiaPlayerService', [
           exploration.isInteractionInline(oldStateName));
 
         if (!_editorPreviewMode) {
+          var isFirstHit = Boolean(visitedStateNames.indexOf(
+            newStateName) === -1);
           StatsReportingService.recordStateTransition(
             oldStateName, newStateName, answer,
-            LearnerParamsService.getAllParams());
+            LearnerParamsService.getAllParams(), isFirstHit);
+          visitedStateNames.push(newStateName);
```

Comment:
This may introduce duplicates in the list. Is that your intention?

Reply:
@seanlip That's not really of any concern really. This is just to know if a new state has never been visited yet.

Reply:
OK; maybe add a one-line comment above the place where this var is declared explaining that "This list may contain duplicates; a state name is added to it each time the learner moves to a new card." 

(It's hard to tell from the name alone whether this allows duplicates or not, since arguably once a state is visited for the first time, it stays visited. But I can't think of a better name, so a comment is probably the next best thing.)

Reply:
Done.